app/components/portfolios/details_component.rb

```diff
@@ -54,10 +54,39 @@ def all_subprojects
       all_descendants.project
     end
 
+    def render_sub_status_bar?
+      # When none of the descendants have a status set, there's nothing to show and we
+      # will return false.
+      sub_statuses.keys.any?(&:present?)
+    end
+
+    def sub_statuses_with_percentages
+      return @status_percentage if @status_percentage
+
+      statuses = sub_statuses
+      total = statuses.values.sum
+
+      @status_percentage = statuses.map do |code, count|
```

Comment:
`sub_statuses` is already memoized, so can be used directly:
```suggestion
      total = sub_statuses.values.sum

      @status_percentage = sub_statuses.map do |code, count|
```